cpp/src/gandiva/function_holder_registry.h

```diff
@@ -66,6 +68,10 @@ class FunctionHolderRegistry {
         {"to_date", LAMBDA_MAKER(ToDateHolder)},
         {"random", LAMBDA_MAKER(RandomGeneratorHolder)},
         {"rand", LAMBDA_MAKER(RandomGeneratorHolder)},
+        {"castINT", LAMBDA_MAKER(ParseStringHolder<arrow::Int32Type>)},
```

Comment:
Using a function holder since according to arrow StringConverter documentation : 
"The class may have a non-trivial construction cost in some cases, so it's recommended to use a single instance many times, if doing bulk conversion"